test/e2e/e2e.go

```diff
@@ -443,6 +443,33 @@ var _ = ginkgo.Describe("e2e control plane", func() {
 	})
 })
 
+// Test pod connectivity to other host IP addresses
+var _ = ginkgo.Describe("test e2e pod connectivity to host addresses", func() {
+	const (
+		ovnWorkerNode    string = "ovn-worker"
+		targetIP         string = "123.123.123.123"
+		svcname          string = "node-e2e-to-host"
+	)
+
+	f := framework.NewDefaultFramework(svcname)
+
+	ginkgo.AfterEach(func() {
+		_, _ = runCommand("docker", "exec", ovnWorkerNode, "ip", "a", "del",
+			fmt.Sprintf("%s/32", targetIP), "dev", "eth0")
+	})
+
+	ginkgo.It("Should validate connectivity from a pod to a non-node host address on same node", func() {
+		// Add another IP address to the worker
+		_, err := runCommand("docker", "exec", ovnWorkerNode, "ip", "a", "add",
+			fmt.Sprintf("%s/32", targetIP), "dev", "eth0")
+		framework.ExpectNoError(err, "failed to add IP to %s", ovnWorkerNode)
+
+		// Spin up another pod that attempts to reach the previously started pod on separate nodes
+		framework.ExpectNoError(
+			checkConnectivityPingToHost(f, ovnWorkerNode, "e2e-src-ping-pod", targetIP, ipv4PingCommand, 30, false))
```

Comment:
Actually this is not a valid test for testing the functionality related to your patch: this works on my kind setup which is not running your patch :)

Reply:
i:e 

```
root@ovn-worker:/# ip a add 172.19.0.24 dev eth0
root@ovn-worker:/# ping 172.19.0.24
PING 172.19.0.24 (172.19.0.24) 56(84) bytes of data.
64 bytes from 172.19.0.24: icmp_seq=1 ttl=64 time=0.097 ms
64 bytes from 172.19.0.24: icmp_seq=2 ttl=64 time=0.093 ms
^C
--- 172.19.0.24 ping statistics ---
```

Reply:
Wait a minute, now that I think back I might have had a local gateway mode cluster. I need to verify

Reply:
Yes, my bad: I originally tested it in local GW mode. This does not work currently in shared. Sorry for the confusion.